planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp

```diff
@@ -333,6 +334,52 @@ class PlannerInterface
     double lpf_gain_dist_to_slow_down;
   };
   SlowDownParam slow_down_param_;
+  struct StopParam
+  {
+    struct ObstacleSpecificParams
+    {
+      double limit_min_acc;
+      double limit_max_dist;
+      bool abandon_to_stop;
+    };
+    const std::unordered_map<uint8_t, std::string> types_maps = {
+      {ObjectClassification::UNKNOWN, "unknown"}, {ObjectClassification::CAR, "car"},
+      {ObjectClassification::TRUCK, "truck"},     {ObjectClassification::BUS, "bus"},
+      {ObjectClassification::TRAILER, "trailer"}, {ObjectClassification::MOTORCYCLE, "motorcycle"},
+      {ObjectClassification::BICYCLE, "bicycle"}, {ObjectClassification::PEDESTRIAN, "pedestrian"}};
+    std::unordered_map<std::string, ObstacleSpecificParams> type_specified_param_list;
+    explicit StopParam(rclcpp::Node & node, const LongitudinalInfo & longitudinal_info)
+    {
+      const std::string param_prefix = "stop.type_specified_params.";
+      const ObstacleSpecificParams default_param{
+        longitudinal_info.limit_min_accel, std::numeric_limits<double>::max(), false};
+      type_specified_param_list.emplace("default", default_param);
+      for (const auto & [type_uint, type_str] : types_maps) {
+        if (node.declare_parameter<bool>(param_prefix + "use_type_specified_params." + type_str)) {
+          const ObstacleSpecificParams param{
+            std::min(
+              node.declare_parameter<double>(param_prefix + "limit_min_acc." + type_str),
+              longitudinal_info.min_accel),
+            node.declare_parameter<double>(param_prefix + "limit_max_dist." + type_str),
+            node.declare_parameter<bool>(param_prefix + "abandon_to_stop." + type_str)};
+          type_specified_param_list.emplace(type_str, param);
+        }
+      }
+    }
+    std::string getParamType(ObjectClassification label)
```

Comment:
Add const reference to the argument, and const to the function. Please check other functions as well.